### DISCUSSION TITLE
Bxx calculator

### DISCUSSION THREAD

**Discussion by thearnieh** [tvtoglu/predictr#5] [General]
Nice library. I am currently trying to find out how to calculate Bxx (or how to use the Bxx calculator). So my question: how can I calculate Bxx-Life (also on the confidence bound(s))?

**Comment by tvtoglu** [comment]
Hi thearnieh! Thanks, glad you like predictr.
The full documentation for the Bx-life calculator is still missing. Mainly because I haven't had the time to fully implement all features I had initially planned. However, the Bx-Life-Percentile Calculator is implemented and you can also get the times for certain unrealibility values from class attributes. Let me show you both ways:

### **Using the Bx-life percentile calculator**
Description: Computes the unreliability at given input time. If you're looking for the the opposit (calculating the time for a given unreliability value, then you can have a look at the work around I am proposing in 2. (see below))

Import predictr
`from predictr import Analysis`

Let's use some data to conduct a Weibull analysis
```
failures = [0.4508831,  0.68564703, 0.76826143, 0.88231395, 1.48287253, 1.62876357]
suspensions = [1.9, 2.0, 2.0]
x = Analysis(df=failures, ds=suspensions, bounds='lrb',show=True)
x.mle()
```
Now we can use the classmethod get_bx_percentile(time, beta, eta)
time is the input time (lifetime, cycle, etc.) for which you want to compute the unreliability
beta is the computed Weibull shape parameter based on your data (stored in your instance x)
eta is the computed Weibull scale parameter based on your data (stored in your instance x)

In my example for the instance x, the classmethod call would look like this for a given time=1.0
```
bx_percentile = Analysis.get_bx_percentile(1.0, x.beta, x.eta)
bx_percentile
```
> 0.3143995

If you use a bias-correction method, e.g. c4 -> use x.beta_c4 and x.eta_c4. You can output the names of attributes for other bias-correction methods using the dir method:
`dir(x)`
This will give you 'beta', 'beta_c4', 'beta_hrbu', 'beta_np_bs', 'beta_p_bs', 'eta', 'eta_c4', 'eta_hrbu', 'eta_np_bs', 'eta_p_bs'. 

Using these for confidence bounds is not complicated, but would involve some coding. Hence, I should really implement it then for easier use ;)


### **Get x-axis values (time, lifetime, cylces, etc.) for a given set of unreliabilty values**
These values are already available and you just need to call them.

The class attribute "unrel" contains [0.001, 0.002, 0.003, 0.005, 0.007, 0.01 , 0.02 , 0.03 , 0.05 , 0.07 , 0.1  , 0.2  , 0.3  , 0.4  , 0.5  , 0.6  , 0.632, 0.7 , 0.8  , 0.9  , 0.95 , 0.99 , 0.999] -> for these unreliability values we cann call the time values of the bounds
bounds_lower and bounds_upper contain the time values.

Remember our instance is called x (see above)
```
vals = dict(percentile=x.unrel, lower_bounds= x.bounds_lower, upper_bound=x.bounds_upper)
list_all_bx = pd.DataFrame(vals)
list_all_bx
```
<img width="369" alt="grafik" src="https://user-images.githubusercontent.com/45008505/205147714-87f8069f-dd93-4a23-8327-60497878087d.png">

Getting the values for the Weibull line would also involve some coding and is not "easy" to use.


As you can see, the implementation is not fully complete and intuitive. I will try to implement and document everything this week. Please give me feedback, if this workaround helped you. If there are some other functionalities missing that you would like to use, please let me know.
